### PR TITLE
feat: send degrees contentful data to taxonomy for skills extraction

### PR DESCRIPTION
**Pull request by AliAdnanSohail**
## Description:

In the follow up this PR https://github.com/openedx/course-discovery/pull/3721. We need to send degrees contentful data to Lightcast to extract skills as well. In this PR we will get aggregated data from contentful and send to lightcast for skills extraction

### CODE REVIEW

File: course_discovery/apps/course_metadata/tests/test_contentful_utils.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def test_transform_bootcamp_contentful_data(self, *args):
         self.assertDictEqual(
             transformed_data, mock_bootcamp_response.bootcamp_transformed_data)
 
-    def test_get_aggregated_data_from_contentful_data(self):
+    def test_get_aggregated_data_from_contentful_data__bootcamp(self):
```
Comment (ansabgillani) [inline course_discovery/apps/course_metadata/tests/test_contentful_utils.py:127]: Can we use `ddt` here instead?

File: course_discovery/apps/course_metadata/tests/test_contentful_utils.py [hunk 2]
```diff
@@ -145,6 +145,24 @@ def test_get_aggregated_data_from_contentful_data(self):
         assert get_aggregated_data_from_contentful_data(
             mock_bootcamp_response.bootcamp_transformed_data, 'test-uuid') == expected_data
 
+    def test_get_aggregated_data_from_contentful_data__degree(self):
```
Comment (ansabgillani) [inline course_discovery/apps/course_metadata/tests/test_contentful_utils.py:148]: Same as above, can we use `ddt` to differentiate degrees and bootcamps?

**Review by ansabgillani — APPROVED** [review]

File: course_discovery/apps/course_metadata/tests/test_contentful_utils.py (hunk 1 above)
Comment (AliAdnanSohail) [reply course_discovery/apps/course_metadata/tests/test_contentful_utils.py:127]: Actually IMO ddt should be used where we are testing a method for some kind of same type but with different input values,
But if type is different then a separate test make more sense to me and make test simpler and readable. Although it make code lengthy but looks simpler, we can get rid of if conditions in test cases

File: course_discovery/apps/course_metadata/tests/test_contentful_utils.py (hunk 2 above)
Comment (AliAdnanSohail) [reply course_discovery/apps/course_metadata/tests/test_contentful_utils.py:148]: same as above

**Review by DawoudSheraz — APPROVED** [review]

**Review by Ali-D-Akbar — APPROVED** [review]

**Pull request merged**